common/src/main/java/org/opensearch/sql/common/utils/QueryContext.java

```diff
@@ -38,8 +39,10 @@ public static void addRequestId() {
    * @return the current request id from {@link ThreadContext}.
    */
   public static String getRequestId() {
-    final String requestId = ThreadContext.get(REQUEST_ID_KEY);
-    return requestId;
+    if (null == ThreadContext.get(REQUEST_ID_KEY)) {
+      addRequestId();
+    }
+    return ThreadContext.get(REQUEST_ID_KEY);
```

Comment:
you can calling ThreadContext.get(REQUEST_ID_KEY) twice.  You can return if the value is !null.